plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2857,9 +2861,49 @@ public ConnectorInsertTableHandle beginRefreshMaterializedView(ConnectorSession
         Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
         beginTransaction(icebergTable);
 
+        fromSnapshotsForRefresh = getFromSnapshotsForIncrementalRefresh(session, sourceTableHandles, icebergTable);
+
         return newWritableTableHandle(table.getSchemaTableName(), icebergTable, retryMode);
     }
 
+    private Map<SchemaTableName, Long> getFromSnapshotsForIncrementalRefresh(ConnectorSession session, List<ConnectorTableHandle> sourceTableHandles, Table icebergTable)
+    {
+        Map<SchemaTableName, Long> fromSnapshotsForRefresh = new HashMap<>();
+        if (!isIncrementalRefreshEnabled(session)) {
+            return fromSnapshotsForRefresh;
+        }
+
+        if (sourceTableHandles.stream().anyMatch(t -> !(t instanceof IcebergTableHandle))) {
+            // Incremental refresh is only supported when all backing tables are Iceberg tables
+            return fromSnapshotsForRefresh;
+        }
+
+        Optional<String> dependencies = Optional.ofNullable(icebergTable.currentSnapshot())
+                .map(Snapshot::summary)
+                .map(summary -> summary.get(DEPENDS_ON_TABLES));
+        if (dependencies.isEmpty() || dependencies.get().equals(UNKNOWN_SNAPSHOT_TOKEN)) {
+            return fromSnapshotsForRefresh;
+        }
+
+        boolean useIncrementalRefresh = !Boolean.parseBoolean(icebergTable.properties().get("materialized-view.incremental-refresh.contains-group-by"))
```

Comment:
what is the purpose of the properties?

Reply:
This is now refactored so that properties are not used anymore